src/asInput/index.jsx

```diff
@@ -89,15 +101,17 @@ const asInput = (WrappedComponent, labelFirst = true) => {
     render() {
       const { description, error, describedBy } = this.getDescriptions();
       return (
-        <div className={styles['form-group']}>
+        <div className={classNames(styles['form-group'])}>
           {labelFirst && <label htmlFor={this.state.id}>{this.props.label}</label>}
           <WrappedComponent
             {...this.props}
             {...this.state}
-            className={[
+            className={[classNames(
               styles['form-control'],
-              ...this.props.className,
-            ]}
+              styles['form-inline'],
+              { [styles['is-invalid']]: !this.state.isValid && this.props.theme.includes('danger') },
```

Comment:
I have no strong feelings about this theme's name. Any opinions?

Reply:
My only thought is that if we ever want to add additional styling in the future whenever `!this.state.isValid` (like always making the background color `red` or something) we'd probably want to name _that_ style `is-invalid`, right?  

But even if we have to rename this class it won't be a breaking change (right?), so I don't think it's a big deal either way.